src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -599,6 +601,8 @@ def _infer_schema_for_file_format(
                     drop_file_format_if_exists_statement(file_format_name)
                 )
             results = self._session._conn.run_query(infer_schema_query)["data"]
+            if len(results) == 0:
+                raise FileNotFoundError(f"Given path: '{path}' could not be found, please ensure filepath exist.")
```

Comment:
Does this happen when the stage path does not exist or when there are no files on the stage?

Reply:
This happen in both situation as long as the filepath is not correct